src/modules/content-seeding/news-aggregator.js

```diff
@@ -0,0 +1,290 @@
+/* flow */
+/* eslint-disable no-console, consistent-this, no-cond-assign */
+
+import feedParser from 'feed-read';
+import _ from 'lodash';
+import { bus, config } from '../../core-server';
+import winston from 'winston';
+import * as pg from '../../lib/pg';
+import promisify from '../../lib/promisify';
+import uuid from 'node-uuid';
+import { TYPE_THREAD, TAG_RSS_NEWS_SEED } from '../../lib/Constants';
+import type { Thread } from '../../lib/schemaTypes';
+
+/*
+	Implementation Details:
+		- TABLES:
+			- FEEDS:
+				column                   type
+			  -----------------------------------------
+			    url		                text
+		        MTBU                    float(24)
+				lastrequesttime         bigint
+				lastupdatetime          bigint
+
+			- ARTICLES:
+				column                   type
+			  -----------------------------------------
+			    url		                text
+		        rawjson                 jsonb
+				terms    		        tsvector
+
+		- ALGORITHM:
+			- Check if the feeds table is empty.
+			- IF empty:
+				- insert a seed url in the feeds table (one time operation).
+			- Truncate the articles table.
+			- Filter feeds where lastrequesttime < ( NOW() - Mean time between updates (MTBU) ).
+			- For each feeds:
+				- Check if the the given feed has some new news article (RSS FEED'S latest article timestamp > lastupdatetime).
+				- If updated, update MTBU, lastrequesttime, lastupdatetime else update lastrequesttime only.
+						MTBU = Average time interval at which a news story is added on that feed.
+				- make a insert query in articles table with url, articleJson, and ts_vector(article_title + article_content).
+				- select roomid, rawjson, url from rooms and articles table where terms lexemes match the room name.
+				- For each result:
+					- Construct a thread with title, url, description from articles and post it into relevent rooms.
+			- Invoke this job every hour.
+*/
+
+
+const performReadQuery = promisify(pg.read.bind(pg, config.connStr));
+const performWriteQuery = promisify(pg.write.bind(pg, config.connStr));
+const parseFeed = promisify(feedParser.bind(feedParser));
+
+type Article = {
+	title: string;
+	link: string;
+	content: string;
+	published: Date;
+}
+
+type FeedsObject = {
+	url: string;
+	mtbu: number;
+	lastrequesttime: number;
+	lastupdatetime: number
+};
+
+type RoomSpecificNews = {
+	roomid: string;
+	roomname: string;
+	article: Article;
+	url: string
+};
+
+function articleAge (articlePubDate: number): number {
+	return Date.now() - articlePubDate;
```

Comment:
To make the function pure, instead of using `Date.now()` inside the function, you can do use default arguments instead,

```js
function articleAge (articlePubDate: number, currentDate: number = Date.now()): number {
    return currentDate - articlePubDate;
}
```

This way, you don't have to pass currentDate, but tests can pass current date to test this.